packages/gatsby-sharp/src/index.ts

```diff
@@ -0,0 +1,52 @@
+const { exec } = require(`child_process`)
+const { createRequire } = require(`module`)
+
+let sharpInstance: typeof import("sharp")
+
+export = async function getSharpInstance(): Promise<typeof import("sharp")> {
+  try {
+    return importSharp()
+  } catch (err) {
+    await rebuildSharp()
+
+    // Try importing again now we have rebuilt sharp
+    return importSharp()
+  }
+}
+
+function importSharp(): typeof import("sharp") {
+  if (!sharpInstance) {
+    const cleanRequire = createRequire(__filename)
+    const sharp = cleanRequire(`sharp`)
+
+    sharp.simd(true)
+    // Concurrency is handled by gatsby
+    sharp.concurrency(1)
+
+    sharpInstance = sharp
+  }
+
+  return sharpInstance
+}
+
+function rebuildSharp(): Promise<string> {
+  return new Promise((resolve, reject) => {
+    exec(
+      `npm rebuild sharp`,
```

Comment:
Does this also work when you use yarn 1/2?

Reply:
yes, I've been doing this trick for 2 years now :joy:.  As longs as npm gets shipped with nodejs we're good.